ESPLoader.js

```diff
@@ -149,23 +153,30 @@ class ESPLoader {
 
         if (wait_response) {
             try {
-                var p = await this.transport.read({timeout: timeout});
-                //console.log("Response " + p);
-                const resp = p[0];
-                const op_ret = p[1];
-                const len_ret = this._bytearray_to_short(p[2], p[3]);
-                const val = this._bytearray_to_int(p[4], p[5], p[6], p[7]);
-                //console.log("Resp "+resp + " " + op_ret + " " + len_ret + " " + val );
-                const data = p.slice(8);
-                if (op == null || op_ret == op) {
-                    return [val, data];
-                } else {
-                    throw("invalid response");
+                for (let i=0 ; i<100 ; i++) {
```

Comment:
Why do we try this 100 times? That seems like it is misrepresenting `timeout`, as now we could end up taking `100*timeout`

Reply:
It won't do 100 timeouts as the first timeout exception will jump to the cat block at line 175.
What this is doing in the try block is reading packets until the packet has a response code of 1, and the opcode is the same as the opcode of the request or the response code is 1 and the data indicates an invalid command. In other cases it will discard the packet and tray again.